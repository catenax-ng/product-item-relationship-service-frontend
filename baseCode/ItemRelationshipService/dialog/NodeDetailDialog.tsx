/********************************************************************************
 * Copyright (c) 2021,2022 BMW Group AG
 * Copyright (c) 2021,2022 Contributors to the CatenaX (ng) GitHub Organisation.
 *
 * See the NOTICE file(s) distributed with this work for additional
 * information regarding copyright ownership.
 *
 * This program and the accompanying materials are made available under the
 * terms of the Apache License, Version 2.0 which is available at
 * https://www.apache.org/licenses/LICENSE-2.0.
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 * License for the specific language governing permissions and limitations
 * under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 ********************************************************************************/

import {
  Dialog,
  DialogContent,
  DialogHeader,
} from 'cx-portal-shared-components'
import { useSelector } from 'react-redux'
import { nodeDialogSelector, getShells } from 'features/irs/slice'
import { ShellDescriptor } from 'features/digitalTwins/types'
import { useTranslation } from 'react-i18next'
import { NodeDetailsTwo } from './NodeDetailsTwo'
import { uniqueId } from 'lodash'

interface NodeDialogProps {
  show: boolean
  onClose: () => void
}

export const NodeDetailDialog = ({ show, onClose }: NodeDialogProps) => {
  const { t } = useTranslation()
  const nodeDialoge = useSelector(nodeDialogSelector)
  const shellList = useSelector(getShells)
  // console.log('shellList', shellList)
  // console.log('nodeDialoge', nodeDialoge)
  const twin = shellList.find(
    (x: ShellDescriptor) => x.globalAssetId.value[0] === nodeDialoge.nodeId
  )
  return (
    <Dialog open={show}>
      <DialogHeader
        title={t('content.irs.dialog.title')}
        closeWithIcon
        onCloseWithIcon={onClose}
      />
      <DialogContent key={uniqueId()}>
        {twin && <NodeDetailsTwo twin={twin} />}
      </DialogContent>
    </Dialog>
  )
}
