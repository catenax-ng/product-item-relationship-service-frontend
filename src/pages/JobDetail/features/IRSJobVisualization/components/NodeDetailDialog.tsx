/********************************************************************************
 * Copyright (c) 2021,2022 BMW Group AG
 * Copyright (c) 2021,2022 Contributors to the CatenaX (ng) GitHub Organisation.
 *
 * See the NOTICE file(s) distributed with this work for additional
 * information regarding copyright ownership.
 *
 * This program and the accompanying materials are made available under the
 * terms of the Apache License, Version 2.0 which is available at
 * https://www.apache.org/licenses/LICENSE-2.0.
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 * License for the specific language governing permissions and limitations
 * under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 ********************************************************************************/

import { Dialog, DialogContent, DialogHeader } from "cx-portal-shared-components";
import uniqueId from "lodash/uniqueId";

import { useTranslation } from "react-i18next";
import { JobResponse, Shell } from "../../../../../types/jobs";
import { NodeDetailsTwo } from "./NodeDetailsTwo";

interface NodeDialogProps {
  showInfo: { shell: Shell; aspectId?: string } | undefined;
  job: JobResponse;
  onClose: () => void;
}

export const NodeDetailDialog = ({ showInfo, onClose, job }: NodeDialogProps) => {
  const { t } = useTranslation();
  const title = `${t("content.irs.dialog.title")}`;

  if (showInfo?.shell === undefined) {
    return null;
  }

  return (
    <Dialog open={showInfo !== undefined}>
      <DialogHeader title={title} closeWithIcon onCloseWithIcon={onClose} />
      <DialogContent key={uniqueId()}>
        <NodeDetailsTwo twin={showInfo.shell} job={job} aspectId={showInfo.aspectId} />
      </DialogContent>
    </Dialog>
  );
};
