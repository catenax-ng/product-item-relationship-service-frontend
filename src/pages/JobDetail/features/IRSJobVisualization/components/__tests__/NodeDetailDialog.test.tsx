import { fireEvent, render, renderWithRouter, screen } from "../../../../../../testing/test-utils";
import { NodeDetailDialog } from "../NodeDetailDialog";

const shellId = "urn:uuid:a45a2246-f6e1-42da-b47d-5c3b58ed62e9";
const shellList = [
  {
    description: [{ language: "en", text: "The shell for a vehicle" }],
    globalAssetId: { value: ["urn:uuid:a45a2246-f6e1-42da-b47d-5c3b58ed62e9"] },
    idShort: "future concept x",
    identification: "882fc530-b69b-4707-95f6-5dbc5e9baaa8",
    specificAssetIds: [{ key: "engineserialid", value: "12309481209312" }],
    submodelDescriptors: [
      {
        description: [{ language: "en", text: "Provides base vehicle information" }],
        endpoints: [
          {
            interface: "HTTP",
            protocolInformation: {
              endpointAddress: "https://catena-x.net/vehicle/basedetails/",
              endpointProtocol: "HTTPS",
              endpointProtocolVersion: "1.0",
            },
          },
        ],
        idShort: "vehicle base details",
        identification: "4a738a24-b7d8-4989-9cd6-387772f40565",
        semanticId: { value: ["urn:bamm:com.catenax.vehicle:0.1.1"] },
      },
      {
        description: [{ language: "en", text: "Provides base vehicle information" }],
        endpoints: [
          {
            interface: "HTTP",
            protocolInformation: {
              endpointAddress: "https://catena-x.net/vehicle/partdetails/",
              endpointProtocol: "HTTPS",
              endpointProtocolVersion: "1.0",
            },
          },
        ],
        idShort: "vehicle part details",
        identification: "dae4d249-6d66-4818-b576-bf52f3b9ae90",
        semanticId: { value: ["urn:bamm:com.catenax.vehicle:0.1.1#PartDetails"] },
      },
    ],
  },
];

it("does not render th node dialog", () => {
  const onClose = jest.fn();
  const { container } = render(<NodeDetailDialog showId={""} shellList={shellList} onClose={onClose} />);
  expect(container).toMatchSnapshot();
});

it("opens the node detail dialog", () => {
  const closeFn = jest.fn();
  renderWithRouter(<NodeDetailDialog showId={shellId} shellList={shellList} onClose={closeFn} />);
  const button = screen.getByRole("button", {
    name: /close/i,
  });
  fireEvent.click(button);
  expect(closeFn).toBeCalledTimes(1);
});
