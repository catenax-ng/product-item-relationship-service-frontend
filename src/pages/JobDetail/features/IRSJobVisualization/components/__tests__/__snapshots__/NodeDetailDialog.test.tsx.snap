// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does not render th node dialog 1`] = `<div />`;
